src/library_manager/lib_manager.c

```diff
@@ -468,16 +467,17 @@ static int lib_manager_load_data_from_host(struct lib_manager_dma_ext *dma_ext,
 	if (ret < 0)
 		return ret;
 
+	ret = dma_get_status(dma_ext->chan->dma->z_dev, dma_ext->chan->index, &stat);
```

Comment:
then you can use:
do
{

}while()

still can have same effect.

Reply:
```
do {
	gather condition
	if (condition met)
		break;

	k_usleep(100);
} while (test for timeout);
```
would work, yes.

Reply:
and to iterate on that:
```
do {
	gather condition
	if (ret < 0 || condition met)
		return ret;

	k_usleep(100);
} while (test for timeout);
```

Reply:
But this is not good either. I don't want needles sleep after the first get_dma_status() and I want the get_dma_status() as the last call at the very end of the loop (at timeout).
this would:
get_dma_status()
sleep
report timout

while in _theory_ the DMA could have finished during the last sleep and we are not checking that, unless we add a get_dma_status() before the report  timeout, which is the same thing as I have right now, but on the other end.

I will not touch this patch.